spec/enum_spec.rb

```diff
@@ -0,0 +1,107 @@
+# This file contains spec tests for Enumerable methods
+
+require '../enum.rb'
+describe Enumerable do
+  let(:empty_array) { [] }
+  let(:array) { [1, 2, 3, 4] }
+  let(:hash) { { 'name' => 'name1' } }
+  let(:range) { (1..4) }
+  describe '#my_each' do
+    it 'return each element of an array' do
+      expect(array.my_each { |element| }).to eql(array.my_each { |element| })
+    end
+    it 'return each element of an hash' do
+      expect(hash.my_each { |element| }).to eql(hash.my_each { |element| })
+    end
+    it 'return each element of a range' do
+      expect(range.my_each { |element| }).to eql(range.my_each { |element| })
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'return each element of an array and its corresponding index' do
+      expect(array.my_each_with_index { |element, index| }).to eql(array.my_each_with_index { |element, index| })
+    end
+    it 'return each element of an hash and its corresponding index' do
+      expect(hash.my_each_with_index { |element, index| }).to eql(hash.my_each_with_index { |element, index| })
+    end
+    it 'return each element of a range and its corresponding index' do
+      expect(range.my_each_with_index { |element, index| }).to eql(range.my_each_with_index { |element, index| })
+    end
+  end
+
+  describe '#my_select' do
+    it 'returns each element for which condition is true' do
+      expect(array.my_select { |i| i > 2 }).to eql([3, 4])
+    end
+  end
+
+  describe '#my_all?' do
+    it 'return true if all of the elements are true for condition' do
+      expect(array.my_all? { |element| element >= 1 }).to eql(true)
+    end
+    it 'return false if all of the elements are false for condition' do
+      expect(array.my_all? { |element| element > 2 }).to eql(false)
+    end
```

Comment:
:bulb:  It statement is `return false if all of the elements are false for condition`. 
Maybe you wanted to say `returns false if the block ever returns false or nil`
Please check the documentation to see&understand how they are verbalizing it => https://ruby-doc.org/core-2.6.5/Enumerable.html